### PR TITLE
Feature/gh 8137 move components to lib

### PR DESCRIPTION
**Pull request by ChristophHi**

### CODE REVIEW

File: feature-libs/product/configurators/common/src/components/service/config-component-test-utils.service.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+
+/**
+ * Configurator component test utils service provides helper functions for the component tests.
+ */
+@Injectable({ providedIn: 'root' })
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/service/config-component-test-utils.service.ts]: Hmm, having this as an `Injectable` doesn't feel like a good practice. 
I'd suggest to remove it, which will effectively make this a regular class.
Moreover, we should try _not_ to export it, if possible. The easiest way to check is to try to import it in the `app.module.ts` from `@spartacus/product`.

Comment (znikola) [reply feature-libs/product/configurators/common/src/components/service/config-component-test-utils.service.ts]: We could move it to the common configurators `/shared/testing`

File: feature-libs/product/configurators/common/src/components/common-configurator-components.module.ts [hunk 2]
```diff
@@ -1,4 +1,39 @@
...
+  exports: [
+    ConfigProductTitleComponent,
+    ConfigTabBarComponent,
+    ConfigPriceSummaryComponent,
+  ],
+  entryComponents: [
+    ConfigProductTitleComponent,
+    ConfigTabBarComponent,
+    ConfigPriceSummaryComponent,
+  ],
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/common-configurator-components.module.ts]: Do you expect to have more common components here?
If so, we can consider having a module per component (in which you can declare and export the component, as well as mark it as an entry component).
Then you could just import those modules here.

Moreover, you could create an `index.ts` file inside of each of the component folders, and export from those index files in https://github.com/SAP/spartacus/pull/8148/files#diff-6bca718ed8bf30e6a313092f450147c5

File: todo.md [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
+
+   - `projects/storefrontapp/src/environments/environment.ts`
+   - `projects/storefrontapp-e2e-cypress/cypress.json`
+   - `projects/storefrontapp-e2e-cypress/cypress.ci.2005.json`
 
 3. Move scss and assets to the `product` lib?
 
 4. Add spinner/👻 design to http://localhost:4200/electronics-spa/en/USD/configureTEXTFIELD/cartEntry/entityKey/0?forceReload=true
+
+5. Get rid of import { makeErrorSerializable } from '@spartacus/product/configurators/common', instead expose it from core lib
```
Comment (znikola) [inline todo.md]: Thanks for this 👍

File: feature-libs/product/configurators/public_api.ts [hunk 4]
```diff
@@ -2,7 +2,8 @@
  * Public API Surface of product configuration
  */
 
+/* 
+export * from '@spartacus/product/configurators/textfield';*/
```
Comment (znikola) [inline feature-libs/product/configurators/public_api.ts]: Discussed offline

File: todo.md (hunk 3 above)
Comment (znikola) [reply todo.md]: To save us some time, maybe also mention the path where it's exposed - `feature-libs/product/configurators/common/src/index.ts`

**Review by znikola — CHANGES_REQUESTED** [review]

File: feature-libs/product/configurators/common/src/components/common-configurator-components.module.ts (hunk 2 above)
Comment (ChristophHi) [reply feature-libs/product/configurators/common/src/components/common-configurator-components.module.ts]: Yes, we expect more components. 
And yes, created modules for each components and dedicated index files

**Review by znikola — APPROVED** [review]

**Pull request merged**